spec/client/ics-007-tendermint-client/README.md

```diff
@@ -184,69 +196,133 @@ function latestClientHeight(clientState: ClientState): Height {
 }
 ```
 
-### Validity predicate
+### VerifyClientMessage
 
 Tendermint client validity checking uses the bisection algorithm described in the [Tendermint spec](https://github.com/tendermint/spec/tree/master/spec/consensus/light-client). If the provided header is valid, the client state is updated & the newly verified commitment written to the store.
 
 ```typescript
-function checkValidityAndUpdateState(
+function verifyClientMessage(
   clientState: ClientState,
-  revision: uint64,
-  header: Header) {
-    // assert revision is correct
-    assert(revision === clientState.currentHeight.revision)
-    // check that revision is encoded correctly in chain ID
-    assert(revision === clientState.chainID.regex('[a-z]*-(0)'))
+  clientMsg: ClientMessage) {
+    switch typeof(clientMsg) {
+      case Header:
+        verifyHeader(clientState, clientMsg)
+      case Misbehaviour:
+        verifyHeader(clientState, clientMsg.h1)
+        verifyHeader(clientState, clientMsg.h2)
```

Comment:
In ibc-go, `verifyMisbehaviour` is called here instead, i.e., https://github.com/cosmos/ibc-go/blob/b6dd7244075b4e56daf4ddcb9b6f7d622eb3a048/modules/light-clients/07-tendermint/update.go#L29